create_lmodrc.py

```diff
@@ -17,6 +17,113 @@
+    local frameStk  = require("FrameStk"):singleton()
+    local mt        = frameStk:mt()
+    local cudaDir = string.gsub(os.getenv('EESSI_SOFTWARE_PATH') or "", 'versions', 'host_injections')
+    local cudaDirExists = exists(cudaDir)
+    if not cudaDirExists then
+        local haveGpu = mt:haveProperty(modT.sn,"arch","gpu")
+        if haveGpu then
+            modT.isVisible = false
+        end
+    end
+end
+
+local function cuda_enabled_load_hook(t)
+    local frameStk  = require("FrameStk"):singleton()
+    local mt        = frameStk:mt()
+    local simpleName = string.match(t.modFullName, "(.-)/")
+    local eprefix = os.getenv('EESSI_PREFIX') .. "/init/gpu_support"
+    -- if we try to load CUDA itself, check if the software exists in host_injections
+    -- otherwise, refuse to load CUDA and print error message
+    if simpleName == 'CUDA' then
+        -- get the full host_injections path
+        local hostInjections = string.gsub(os.getenv('EESSI_SOFTWARE_PATH') or "", 'versions', 'host_injections')
+        -- build final path where the CUDA software should be installed
+        local cudaEasyBuildDir = hostInjections .. "/software/" .. t.modFullName .. "/easybuild"
+        local cudaDirExists = exists(cudaEasyBuildDir)
+        if not cudaDirExists then
+            io.stderr:write("You requested to load ",simpleName,"\\n")
+            io.stderr:write("While the module file exists, the actual software is not shipped with EESSI.\\n")
+            io.stderr:write("In order to be able to use the CUDA module, please follow the instructions in the\\n")
+            io.stderr:write("gpu_support folder. Adding the CUDA software can be as easy as:\\n")
+            io.stderr:write("export INSTALL_CUDA=true && ./add_nvidia_gpu_support.sh\\n")
+            frameStk:__clear()
+        end
+    end
+    -- when loading CUDA enabled modules check if the necessary matching compatibility libraries are installed
+    -- otherwise, refuse to load the requested module and print error message
+    local haveGpu = mt:haveProperty(simpleName,"arch","gpu")
+    if haveGpu then
+        local arch = os.getenv("EESSI_CPU_FAMILY") or ""
+        local cudaVersionFile = "/cvmfs/pilot.eessi-hpc.org/host_injections/nvidia/" .. arch .. "/latest/version.txt"
+        local cudaDriverExists = exists(cudaVersionFile)
+        local singularityCudaExists = exists("/.singularity.d/libs/libcuda.so")
+        if not (cudaDriverExists or singularityCudaExists)  then
+            io.stderr:write("You requested to load ",simpleName,"\\n")
+            io.stderr:write("which relies on the CUDA runtime environment and its compatibility libraries.\\n")
+            io.stderr:write("In order to be able to use the module, please follow the instructions in the\\n")
+            io.stderr:write("gpu_support folder. Installing the needed compatibility libraries can be as easy as:\\n")
+            io.stderr:write("./add_nvidia_gpu_support.sh\\n")
```

Comment:
This should be reworded a bit, there is no script `add_nvidia_gpu_support.sh` yet, and the compat libraries are only needed if the GPU driver is too old.

Reply:
I would reword this to point to https://eessi.io/docs/gpu for now, see also https://github.com/EESSI/docs/pull/128

Reply:
Pointing to that doc page for now